cmake/compiler_flag_sets/native.cmake

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# C, CXX, LD, and AS flags for native targets.
+#
+
+#
+# Debug flags for C and C++
+#
+set(C_FLAG_SET
+                "-fprofile-arcs"
+                "-ftest-coverage"
+                "-Og"
+                "-DDEBUG"
+                "-ggdb"
+)
+
+#
+# Diagnostics for C and C++
+#
+list(APPEND C_FLAG_SET
+                "-pedantic"
+                "-Wall"
+                "-Wextra"
+                "-Werror"
+                "-Wfloat-equal"
+                "-Wconversion"
+                "-Wabi"
+                "-Wunused-parameter"
+                "-Wunused-variable"
+                "-Wunused-value"
+                "-Wcast-align"
+                "-Wmissing-declarations"
+                "-Wno-missing-field-initializers"
+                "-Wdouble-promotion"
+                "-Wswitch-enum"
+                "-Wtype-limits"
+                "-Wno-error=array-bounds"
+)
+
+set(CXX_FLAG_SET ${C_FLAG_SET})
+set(ASM_FLAG_SET ${C_FLAG_SET})
+
+#
+# General C-only flags
```

Comment:
Do we actually need those?

Reply:
I think so. The way cmake works is these are keyed on the project's supported languages so you need flags for each language you declare.

Reply:
My point is that we don't really have any C code here, do we?

Reply:
ah. okay. will remove this from the native set. thanks.